technology-considerations/TDR Database Design.md

```diff
@@ -0,0 +1,320 @@
+| ----- | ----- | ----- | ----- | ----- |  
+| PropertyId |   Internal ID of the property  | PK             
+|  Name |         The full name of the property |  String |                              
+|  Description |  The description of the property | String |                              
+|  Shortname |    The short name, code or label of the property |  String | | To be used in creating the export column header
+
+### AVMetadata
+
+The AVMetadata table is proposed to hold information gathered during
+anti-virus analysis of a file.
+
+The columns suggested are to capture the software and versions used to
+analysis the files. If a file is to be analysed by multiple anti-virus
+packages it is suggested a row be added for each analysis.
+
+The exact nature of the Result columns used to capture the result of the
+analysis will need to be discussed further since the nature or style of
+results may vary between anti-virus software packages.
+
+| Field | Description | Data Value type | Key Type | Comment |
+| ----- | ----- | ----- | ----- | ----- |  
+|  FileId |           The ID of the file analysed || FK             
+|  Software | | String                               
+|  SoftwareVersion | | String                               
+|  DatabaseVersion ||  String                               
+|  Result | | String  ||                             The result returned... Need to clarify what is returned.. may need more columns
+|  Datetime |         Time stamp of row creation  |  datetime                             
+
+### ClientFileMetadata
+
+The ClientFileMetadata table is suggested to hold information gathered
+during initial client side analysis before the files are uploaded.
+
+The checksum and algorithm used will need to be captured for file
+verification later in the TDR process. Should multiple checksums be
+created then more than one row can exist for a file.
+
+The original path is required as the TDR export package (bagIt Bag) will
+need to be created using the same folder hierarchal structure and the
+files placed at the appropriate place.
+
+The use of specific column names within the table, rather than the
+Key:Value style used in the FileMetadata table is suggested since the
+properties captured are unlikely to change very much.
+
+| Field | Description | Data Value type | Key Type | Comment |
+| ----- | ----- | ----- | ----- | ----- | 
+| FileId      | The ID of the file analysed |             | FK          |             |
+| OriginalPath | The full file path of the original file| String      |             | Or as much as we can get. i.e. starting from top level folder that  was selected to transfer
+| Checksum    | The value of the checksum  | String      |             |             |
+| ChecksumType | sha256, md5 | String      |             |             |
+| LastModifiedDate  |             | datetime    |             |             |
+| CreationDate |             | datetime    |             |             |
```

Comment:
I don't think we can get the creation date on the client side. We only have the last modified date. See the [Mozilla File API docs](https://developer.mozilla.org/en-US/docs/Web/API/File#Properties) for the properties we can read in the browser.

We might be able to get the creation date for some files using something like Apache Tika, but it won't be client-side data.

Reply:
It was something that was asked for but I did realise we may not be able to get it. I think it should stay in the doc but may not be put into the app. Perhaps a technical discussion with the business to look at the value of having the data versus implementation cost and time. Does that make sense?

Reply:
I think you're right that we should have a discussion about how critical the creation date is, and how we can get it if not through JavaScript. Would you mind adding a note to this table so it's clear that we won't (almost certainly) won't be able to get it in the browser?

Reply:
I've added a similar comment to that added the  export document

Reply:
Thanks - I see the comment about the file path. Could you add a similar one about the creation date as well?